platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheGradlePropertiesIntegrationTest.groovy

```diff
@@ -141,7 +164,7 @@ class ConfigurationCacheGradlePropertiesIntegrationTest extends AbstractConfigur
         then:
         outputContains '2!'
         configurationCache.assertStateStored()
-        outputContains "because the set of Gradle properties has changed."
+        outputContains "because the set of Gradle properties has changed: values for [gradleProp] were changed, [unusedProperty] were added."
```

Comment:
💄 This is definitely an improvement but it would be nicer to keep with the prose spirit of the messages and display something like _the value of 'gradleProp' has changed and, 'usedProperty' was added_. In the case of multiple properties: _the values of 'gradleProp1', 'gradleProp2' and 'gradleProp3' were changed and, 'usedProp1' and 'usedProp2' were added.